### DISCUSSION TITLE
Modify max response body size using Envoy

### DISCUSSION THREAD

**Discussion by sfontana9** [mia-platform/community#186] [Q&A] [answered]
Hi,
I would like to modify the default maximum response body size (4KB) of my Envoy api gateway.
I can see [here](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route.proto) that Envoy allows a **max_direct_response_body_size_bytes** parameter in its HTTP Route Configuration. What's the best way to access this value using Mia-Platform console's advanced configurations?

Thank you

**Comment by mattia-fumo** [comment, marked as answer]
Hi @sfontana9,
you could use the [property patch extension](https://docs.mia-platform.eu/docs/development_suite/api-console/advanced-section/api-gateway-envoy/extensions#patching-arbitrary-listeners-properties) like this:
```yaml
- listener_name: <your listener>
  'filter_chains.0.filters.0.typed_config.route_config.max_direct_response_body_size_bytes': <your body size>
```

Please, notice that this property should be changed with caution, as reported [here](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route.proto):
```
Envoy currently holds the content of direct response body in memory. 
Be careful setting this to be larger than the default 4KB, since the allocated 
memory for direct response body is not subject to data plane buffering controls.
```

**Reply by mattia-fumo** [reply]
Also notice, that this property will only affect direct response configuration. That is responses directly managed by envoy like local replies
